--- origsrc/epiphany-extensions-2.22.2/configure.ac	2008-05-26 13:08:15.000000000 -0500
+++ src/epiphany-extensions-2.22.2/configure.ac	2008-06-02 23:06:43.937500000 -0500
@@ -365,6 +365,23 @@
 
 GECKO_DEFINES
 
+# *****
+# Win32
+# *****
+
+case "$host_os" in
+	cygwin*)
+		platform_win32=yes
+		no_undefined="-no-undefined"
+		;;
+	*)
+		platform_win32=no
+		no_undefined=
+		;;
+esac
+AC_SUBST(no_undefined)
+AM_CONDITIONAL(PLATFORM_WIN32, test "$platform_win32" = "yes")
+
 # *****************
 # Add warning flags
 # *****************
--- origsrc/epiphany-extensions-2.22.2/extensions/actions/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/actions/Makefile.am	2008-06-02 22:29:44.734375000 -0500
@@ -22,8 +22,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libactionsextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libactionsextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/adblock/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/adblock/Makefile.am	2008-06-02 23:17:39.906250000 -0500
@@ -27,10 +27,11 @@
 	$(AM_CFLAGS)
 
 libadblockextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)	\
 	$(PCRE_LIBS)
 
 libadblockextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/auto-reload/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/auto-reload/Makefile.am	2008-06-02 22:30:37.031250000 -0500
@@ -6,8 +6,11 @@
 	ephy-auto-reload-extension.h	\
 	extension.c
 	
+libautoreloadextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libautoreloadextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/auto-scroller/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/auto-scroller/Makefile.am	2008-06-02 22:57:35.156250000 -0500
@@ -20,10 +20,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
-libautoscrollerextension_la_LIBADD =
+libautoscrollerextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
 
 libautoscrollerextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/certificates/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/certificates/Makefile.am	2008-06-02 22:31:15.250000000 -0500
@@ -18,8 +18,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libcertificatesextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libcertificatesextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/error-viewer/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/error-viewer/Makefile.am	2008-06-02 22:31:47.734375000 -0500
@@ -41,6 +41,7 @@
 
 liberrorviewerextension_la_LIBADD = \
 	mozilla/liberrorviewermozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 if HAVE_OPENSP
@@ -50,7 +51,7 @@
 endif
 
 liberrorviewerextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/extensions-manager-ui/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/extensions-manager-ui/Makefile.am	2008-06-02 22:32:10.421875000 -0500
@@ -18,8 +18,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libextensionsmanageruiextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libextensionsmanageruiextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/gestures/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/gestures/Makefile.am	2008-06-02 22:32:34.390625000 -0500
@@ -20,8 +20,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libgesturesextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libgesturesextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
  
--- origsrc/epiphany-extensions-2.22.2/extensions/greasemonkey/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/greasemonkey/Makefile.am	2008-06-02 22:32:54.421875000 -0500
@@ -24,6 +24,7 @@
 
 libgreasemonkeyextension_la_LIBADD = \
 	mozilla/libgreasemonkeymozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)		\
 	$(PCRE_LIBS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/java-console/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/java-console/Makefile.am	2008-06-02 22:59:10.906250000 -0500
@@ -21,10 +21,11 @@
 
 libjavaconsoleextension_la_LIBADD = \
 	mozilla/libjavaconsolemozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libjavaconsoleextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/livehttpheaders/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/livehttpheaders/Makefile.am	2008-06-02 22:59:32.843750000 -0500
@@ -26,8 +26,13 @@
 
 liblivehttpheadersextension_la_LIBADD = \
 	mozilla/liblivehttpheadersmozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
-liblivehttpheadersextension_la_LDFLAGS = -module -avoid-version
+
+liblivehttpheadersextension_la_LDFLAGS = \
+	-module -avoid-version $(no_undefined) \
+	-export-symbols $(top_srcdir)/ephy-extension.symbols \
+	$(AM_LDFLAGS)
 
 extensioninidir = $(extensiondir)
 extensionini_in_files = livehttpheaders.ephy-extension.in.in
--- origsrc/epiphany-extensions-2.22.2/extensions/page-info/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/page-info/Makefile.am	2008-06-02 22:39:17.187500000 -0500
@@ -24,10 +24,11 @@
 
 libpageinfoextension_la_LIBADD = \
 	mozilla/libpageinfomozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libpageinfoextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/permissions/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/permissions/Makefile.am	2008-06-02 22:39:42.843750000 -0500
@@ -18,8 +18,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libpermissionsextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libpermissionsextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/push-scroller/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/push-scroller/Makefile.am	2008-06-02 22:59:55.109375000 -0500
@@ -19,10 +19,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
-libpushscrollerextension_la_LIBADD =
+libpushscrollerextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
 
 libpushscrollerextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/rss/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/rss/Makefile.am	2008-06-02 22:40:51.453125000 -0500
@@ -20,8 +20,12 @@
 	$(DBUS_CFLAGS) \
 	$(AM_CFLAGS)
 
+librssextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
+	$(DBUS_LIBS)
+
 librssextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/sample/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/sample/Makefile.am	2008-06-02 22:51:11.843750000 -0500
@@ -16,8 +16,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libsampleextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libsampleextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/sample-mozilla/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/sample-mozilla/Makefile.am	2008-06-02 22:50:42.984375000 -0500
@@ -21,10 +21,11 @@
 
 libsamplemozillaextension_la_LIBADD = \
 	mozilla/libsamplemozillamozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libsamplemozillaextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/select-stylesheet/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/select-stylesheet/Makefile.am	2008-06-02 22:51:29.390625000 -0500
@@ -23,10 +23,11 @@
 
 libselectstylesheetextension_la_LIBADD = \
 	mozilla/libselectstylesheetmozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libselectstylesheetextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/sidebar/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/sidebar/Makefile.am	2008-06-02 22:54:58.421875000 -0500
@@ -22,8 +22,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libsidebarextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libsidebarextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/smart-bookmarks/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/smart-bookmarks/Makefile.am	2008-06-02 22:56:21.390625000 -0500
@@ -21,10 +21,11 @@
 
 libsmartbookmarksextension_la_LIBADD =	\
 	$(top_builddir)/extensions/smart-bookmarks/mozilla/libsbmozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libsmartbookmarksextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/tab-groups/Makefile.am	2008-03-05 13:02:59.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/tab-groups/Makefile.am	2008-06-02 22:56:45.937500000 -0500
@@ -17,8 +17,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libtabgroupsextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libtabgroupsextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.22.2/extensions/tab-states/Makefile.am	2008-03-05 13:03:00.000000000 -0600
+++ src/epiphany-extensions-2.22.2/extensions/tab-states/Makefile.am	2008-06-02 22:57:05.156250000 -0500
@@ -16,8 +16,11 @@
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libtabstatesextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libtabstatesextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
