--- origsrc/epiphany-extensions/configure.ac	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/configure.ac	2009-06-25 19:20:00.513633900 -0500
@@ -239,6 +239,23 @@ if echo "$EXTENSIONS" | egrep '(^| )(rss
 	PKG_CHECK_MODULES([DBUS], [dbus-1 >= $DBUS_REQUIRED dbus-glib-1 >= $DBUS_GLIB_REQUIRED])
 fi
 
+# *****
+# Win32
+# *****
+
+case "$host_os" in
+	cygwin*)
+		platform_win32=yes
+		no_undefined="-no-undefined"
+		;;
+	*)
+		platform_win32=no
+		no_undefined=
+		;;
+esac
+AC_SUBST(no_undefined)
+AM_CONDITIONAL(PLATFORM_WIN32, test "$platform_win32" = "yes")
+
 # *****************
 # Add warning flags
 # *****************
--- origsrc/epiphany-extensions/extensions/actions/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/actions/Makefile.am	2009-06-25 19:20:00.516133900 -0500
@@ -22,8 +22,11 @@ libactionsextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libactionsextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libactionsextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/adblock/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/adblock/Makefile.am	2009-06-25 19:20:28.774065700 -0500
@@ -25,8 +25,11 @@ libadblockextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libadblockextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libadblockextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/auto-reload/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/auto-reload/Makefile.am	2009-06-25 19:21:13.703502200 -0500
@@ -6,8 +6,11 @@ libautoreloadextension_la_SOURCES = \
 	ephy-auto-reload-extension.h	\
 	extension.c
 	
+libautoreloadextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libautoreloadextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/auto-scroller/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/auto-scroller/Makefile.am	2009-06-25 19:20:00.523634100 -0500
@@ -20,10 +20,11 @@ libautoscrollerextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
-libautoscrollerextension_la_LIBADD =
+libautoscrollerextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
 
 libautoscrollerextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/certificates/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/certificates/Makefile.am	2009-06-25 19:20:00.527384100 -0500
@@ -18,8 +18,11 @@ libcertificatesextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libcertificatesextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libcertificatesextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/error-viewer/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/error-viewer/Makefile.am	2009-06-25 19:20:00.528634100 -0500
@@ -41,6 +41,7 @@ liberrorviewerextension_la_CFLAGS = \
 
 liberrorviewerextension_la_LIBADD = \
 	mozilla/liberrorviewermozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 if HAVE_OPENSP
@@ -50,7 +51,7 @@ liberrorviewerextension_la_LIBADD += \
 endif
 
 liberrorviewerextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/extensions-manager-ui/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/extensions-manager-ui/Makefile.am	2009-06-25 19:20:00.531134200 -0500
@@ -18,8 +18,11 @@ libextensionsmanageruiextension_la_CFLAG
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libextensionsmanageruiextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libextensionsmanageruiextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/gestures/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/gestures/Makefile.am	2009-06-25 19:20:00.533634200 -0500
@@ -20,8 +20,11 @@ libgesturesextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libgesturesextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libgesturesextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
  
--- origsrc/epiphany-extensions/extensions/greasemonkey/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/greasemonkey/Makefile.am	2009-06-25 19:22:33.043464500 -0500
@@ -21,11 +21,12 @@ libgreasemonkeyextension_la_CFLAGS = \
 	$(AM_CFLAGS)
 
 libgreasemonkeyextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(WEBKIT_LIBS)		\
 	$(PCRE_LIBS)
 
 libgreasemonkeyextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/java-console/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/java-console/Makefile.am	2009-06-25 19:20:00.539884300 -0500
@@ -21,10 +21,11 @@ libjavaconsoleextension_la_CFLAGS = \
 
 libjavaconsoleextension_la_LIBADD = \
 	mozilla/libjavaconsolemozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libjavaconsoleextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/livehttpheaders/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/livehttpheaders/Makefile.am	2009-06-25 19:20:00.542384300 -0500
@@ -26,8 +26,13 @@ glade_DATA = livehttpheaders-ui.glade
 
 liblivehttpheadersextension_la_LIBADD = \
 	mozilla/liblivehttpheadersmozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
-liblivehttpheadersextension_la_LDFLAGS = -module -avoid-version
+
+liblivehttpheadersextension_la_LDFLAGS = \
+	-module -avoid-version $(no_undefined) \
+	-export-symbols $(top_srcdir)/ephy-extension.symbols \
+	$(AM_LDFLAGS)
 
 extensioninidir = $(extensiondir)
 extensionini_in_files = livehttpheaders.ephy-extension.in.in
--- origsrc/epiphany-extensions/extensions/page-info/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/page-info/Makefile.am	2009-06-25 19:20:00.544884400 -0500
@@ -24,10 +24,11 @@ libpageinfoextension_la_CFLAGS = \
 
 libpageinfoextension_la_LIBADD = \
 	mozilla/libpageinfomozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libpageinfoextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/permissions/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/permissions/Makefile.am	2009-06-25 19:20:00.546134400 -0500
@@ -18,8 +18,11 @@ libpermissionsextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libpermissionsextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libpermissionsextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/push-scroller/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/push-scroller/Makefile.am	2009-06-25 19:20:00.548634400 -0500
@@ -19,10 +19,11 @@ libpushscrollerextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
-libpushscrollerextension_la_LIBADD =
+libpushscrollerextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
 
 libpushscrollerextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/rss/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/rss/Makefile.am	2009-06-25 19:20:00.551134500 -0500
@@ -20,8 +20,12 @@ librssextension_la_CFLAGS = \
 	$(DBUS_CFLAGS) \
 	$(AM_CFLAGS)
 
+librssextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
+	$(DBUS_LIBS)
+
 librssextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/sample/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/sample/Makefile.am	2009-06-25 19:20:00.553634500 -0500
@@ -16,8 +16,11 @@ libsampleextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libsampleextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libsampleextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/select-stylesheet/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/select-stylesheet/Makefile.am	2009-06-25 19:20:00.556134600 -0500
@@ -23,10 +23,11 @@ libselectstylesheetextension_la_CFLAGS =
 
 libselectstylesheetextension_la_LIBADD = \
 	mozilla/libselectstylesheetmozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libselectstylesheetextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/sidebar/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/sidebar/Makefile.am	2009-06-25 19:20:00.558634600 -0500
@@ -22,8 +22,11 @@ libsidebarextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libsidebarextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libsidebarextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/smart-bookmarks/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/smart-bookmarks/Makefile.am	2009-06-25 19:20:00.561134600 -0500
@@ -21,10 +21,11 @@ libsmartbookmarksextension_la_CFLAGS = \
 
 libsmartbookmarksextension_la_LIBADD =	\
 	$(top_builddir)/extensions/smart-bookmarks/mozilla/libsbmozilla.la	\
+	$(EPIPHANY_DEPENDENCY_LIBS)		\
 	$(MOZILLA_COMPONENT_LIBS)
 
 libsmartbookmarksextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions-2.27.4/extensions/soup-fly/Makefile.am	2009-07-01 05:57:15.000000000 -0500
+++ src/epiphany-extensions-2.27.4/extensions/soup-fly/Makefile.am	2009-09-08 22:15:39.026424200 -0500
@@ -18,8 +18,11 @@ libsoupflyextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libsoupflyextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libsoupflyextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version -no-undefined \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/tab-groups/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/tab-groups/Makefile.am	2009-06-25 19:20:00.563634700 -0500
@@ -17,8 +17,11 @@ libtabgroupsextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libtabgroupsextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libtabgroupsextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
--- origsrc/epiphany-extensions/extensions/tab-states/Makefile.am	2009-06-25 19:11:14.000000000 -0500
+++ src/epiphany-extensions/extensions/tab-states/Makefile.am	2009-06-25 19:20:00.566134700 -0500
@@ -16,8 +16,11 @@ libtabstatesextension_la_CFLAGS = \
 	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
 	$(AM_CFLAGS)
 
+libtabstatesextension_la_LIBADD = \
+	$(EPIPHANY_DEPENDENCY_LIBS)
+
 libtabstatesextension_la_LDFLAGS = \
-	-module -avoid-version \
+	-module -avoid-version $(no_undefined) \
 	-export-symbols $(top_srcdir)/ephy-extension.symbols \
 	$(AM_LDFLAGS)
 
